pkg/secrethub/client.go

```diff
@@ -54,12 +54,12 @@ type ClientInterface interface {
 var (
 	errClient = errio.Namespace("client")
 
-	whitelistAppInfoName = regexp.MustCompile("^[a-zA-Z0-9_-]*$")
+	whitelistAppInfoName = regexp.MustCompile("^[a-zA-Z0-9_-]{2,64}$")
 )
 
 // Errors
 var (
-	ErrInvalidAppInfoName = errClient.Code("invalid_app_info_name").Error("name must be set and can only contain alphanumeric, underscore (_), and dash (-) characters")
+	ErrInvalidAppInfoName = errClient.Code("invalid_app_info_name").Error("name must be 2-64 characters long, only alphanumeric, underscore (_), and dash (-)")
```

Comment:
As a reference, 64 characters is this long: `Lorem ipsum dolor sit amet, consectetur adipiscing elit posuerer`. Is there any valid reason for such a long name?

If we make it 32 now, we can later always increase it. In the meantime, it forces people to give the name some thought and make it (relatively) conise (32 is still a lot of characters).

Reply:
The reason I made it longer is that sometimes platforms require a specific naming format for plugins (e.g. https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin). But 64 is indeed somewhat on the high end, so I'll make it 50.